packages/front-end/components/HomePage/NorthStarMetricDisplay.tsx

```diff
@@ -24,64 +37,166 @@ const NorthStarMetricDisplay = ({
   hoverDate?: number | null;
   onHoverCallback?: (ret: { d: number | null }) => void;
 }): React.ReactElement => {
-  const { project, getDatasourceById } = useDefinitions();
+  const {
+    project,
+    getDatasourceById,
+    getExperimentMetricById,
+    getFactTableById,
+  } = useDefinitions();
+
+  const orgSettings = useOrgSettings();
   const permissionsUtil = usePermissionsUtil();
   const { apiCall } = useAuth();
 
+  const metric = getExperimentMetricById(metricId);
+
   const { data, error, mutate } = useApi<{
-    metric: MetricInterface;
-    experiments: DraftExperiment[];
-  }>(`/metric/${metricId}`);
+    data: {
+      experiments: ExperimentInterfaceStringDates[];
+      analysis: MetricAnalysisInterface | null;
+      metric: MetricInterface | null;
+    };
+  }>(`/metrics/${metricId}/northstar`);
 
-  // @todo: get the metric period in days from the 'window'.
+  if (!metric || !data) {
+    return <LoadingOverlay />;
+  }
```

Comment:
You typically want to check for `error` first before the loading state.  When there's an error, data is often empty, so with how it is here it would show a loading overlay forever.

Reply:
done ty